spring-cloud-task-core/src/main/java/org/springframework/cloud/task/configuration/DefaultTaskConfigurer.java

```diff
@@ -45,64 +41,70 @@
  * </ul>
  *
  * @author Glenn Renfro
+ * @author Michael Minella
  */
 public class DefaultTaskConfigurer implements TaskConfigurer {
 
-	private final static Logger logger = LoggerFactory.getLogger(DefaultTaskConfigurer.class);
-
-	private DataSource dataSource;
-
 	private TaskRepository taskRepository;
 
 	private TaskExplorer taskExplorer;
 
 	private PlatformTransactionManager transactionManager;
 
-	public DefaultTaskConfigurer(){
-		initialize();
-	}
+	private ConfigurableApplicationContext context;
 
-	public DefaultTaskConfigurer(DataSource dataSource) {
-		this.dataSource = dataSource;
-		initialize();
+	private TaskExecutionDaoFactoryBean taskExecutionDaoFactoryBean;
+
+	public DefaultTaskConfigurer(ConfigurableApplicationContext context) {
+		this.context = context;
 	}
 
 	@Override
 	public TaskRepository getTaskRepository() {
-		return taskRepository;
+		if(this.taskRepository == null) {
+			if(this.taskExecutionDaoFactoryBean == null) {
+				this.taskExecutionDaoFactoryBean = new TaskExecutionDaoFactoryBean(this.context);
+			}
+
+			this.taskRepository = new SimpleTaskRepository(this.taskExecutionDaoFactoryBean);
+		}
+
+		return this.taskRepository;
 	}
 
 	@Override
 	public TaskExplorer getTaskExplorer() {
+		if(this.taskExplorer == null) {
+			if(this.taskExecutionDaoFactoryBean == null) {
+				this.taskExecutionDaoFactoryBean = new TaskExecutionDaoFactoryBean(this.context);
```

Comment:
Why not create the taskExecutionDAoFactoryBean in the ctor? *same comment as above*